src/main/kotlin/provider/btc/BtcPublicKeyProvider.kt

```diff
@@ -30,6 +43,48 @@ class BtcPublicKeyProvider(
         }
     }
 
+    fun checkAndCreateMsAddress(notaryKeys: Collection<String>): Result<Unit, Exception> {
+        return Result.of {
+            val peers = notaryPeerListProvider.getPeerList().size
+            if (peers == 0) {
+                throw IllegalStateException("no peers to create btc multi signature address")
+            } else if (notaryKeys.size == peers && hasMyKey(notaryKeys)) {
```

Comment:
Maybe we could throw something for missing else branch?

Reply:
no, this is a regular situation. not having enough keys to create multisignature BTC address means that we need more keys to proceed.